vernac/obligations.ml

```diff
@@ -223,7 +189,9 @@ and solve_obligation_by_tac prg obls i tac =
           if def && not prg.prg_poly then (
             (* Declare the term constraints with the first obligation only *)
             let evd = Evd.from_env (Global.env ()) in
-            let evd = Evd.merge_universe_subst evd (Evd.universe_subst (Evd.from_ctx ctx)) in
+            let evd =
+              Evd.merge_universe_subst evd (Evd.universe_subst (Evd.from_ctx ctx))
```

Comment:
btw here we can just do
```suggestion
              Evd.merge_universe_subst evd (UState.subst ctx)
```

Reply:
Good catch, done thanks.

Reply:
Note that Evd.merge_universe_subst has become unused after this change; maybe should be removed?

Reply:
>Note that Evd.merge_universe_subst has become unused after this change; maybe should be removed?

Sure why not

Reply:
Ok, as this has missed 8.12 will add that commit [providing equations doesn't use it]

Reply:
Finally I did some more cleaning that I had planned, see #12354